### ISSUE TITLE
Source map line numbers wrong

### ISSUE THREAD

**Issue by Floffah** [egoist/tsup#434]
This issue is the same as the one in esbuild [#638](https://github.com/evanw/esbuild/issues/638) and was fixed there. I have tested my codebase using raw esbuild and the issue is not present there.
Basically, the problem is that line numbers are wrong in source maps therefore in stack traces.

Expected (using esbuild):
![image](https://user-images.githubusercontent.com/27270386/136701921-f4ac6d12-ccf1-49fb-a90e-857a766c81fb.png)
![image](https://user-images.githubusercontent.com/27270386/136701930-68171920-3834-4b9f-85fd-4f06ee08a1ba.png)
(i know i forgot to minify, however if i minify it the line number is still correct here)

What really happens (with tsup):
![image](https://user-images.githubusercontent.com/27270386/136701982-e267caf6-2449-4eea-b671-be733376ecf8.png)
![image](https://user-images.githubusercontent.com/27270386/136701991-73713731-1b31-46f0-aed8-04e950ac0b1a.png)

**Comment by alexng353** [comment]
Hey I'm having the same error on version 7.2.0. I'm using the source map visualizer provided by the referenced issue in esbuild. There are definitely some problems, but I can't out how this stuff works. Posting to bump this issue.

![image](https://github.com/egoist/tsup/assets/26424336/10be8036-1e03-44cf-b028-dead4f0742de)

**Comment by alexng353** [comment]
right, the issue in esbuild ([#638](https://github.com/evanw/esbuild/issues/638)) mentions that this bug is fixed (?) in esbuild 0.8.30 and tsup is on 0.8.12 (below). Their issue mentions that it should be fine <=0.8.12, >=0.8.30 but as shown in the image above it is definitely not fine.

https://github.com/egoist/tsup/blob/9b813620a884f5b780346a240c51ead321227bce/package.json#L35C1-L36C1
